Code/Framework/AzCore/AzCore/Asset/AssetContainer.cpp

```diff
@@ -232,8 +233,20 @@ namespace AZ::Data
         // Finally, after creating and queueing the dependent assets, queue the root asset.  This is saved until last to ensure that
         // it doesn't have any chance of serializing in until after all the dependent assets have been queued for loading and have
         // been added to the list of dependencies.
-        auto thisAsset = AssetManager::Instance().GetAssetInternal(rootAssetId, rootAssetType, rootAsset.GetAutoLoadBehavior(),
-            loadParamsCopyWithNoLoadingFilter, AssetInfo(), HasPreloads(rootAssetId));
+        Asset<AssetData> thisAsset;
+
+        if (!m_isReload)
```

Comment:
might want to add some comments to explain why reload need special handling here.

Reply:
Added